theme/static/js/generic-resource.js

```diff
@@ -222,34 +227,38 @@ function showAddEditExtraMetaPopup(edit, row_id_str) {
     $('#extraMetaDialog').modal('show');
 }
 
-
 function addEditExtraMeta2Table() {
+    // Restore validation UI state
     $("#extra_meta_msg").hide();
+    $("#extra_meta_name_input").removeClass("form-invalid");
     var t = $('#extraMetaTable').DataTable();
     var extra_meta_name = $("#extra_meta_name_input").val().trim();
     var extra_meta_value = $("#extra_meta_value_input").val().trim();
     var edit_extra_meta_row_id = $("#edit_extra_meta_row_id").val().trim();
 
-    if(extra_meta_name.length==0 || extra_meta_value.length==0) {
-        $("#extra_meta_msg").html("<font color='red'>Both name and value are required fields that cannot be left blank.</font>");
+    if (foundDuplicatedName(t, extra_meta_name, edit_extra_meta_row_id)) {
+        $("#extra_meta_name_input").addClass("form-invalid");
+        $("#extra_meta_msg").html("<div class='alert alert-danger'>" +
+            "The name already exists. Please input a different name.</div>");
         $("#extra_meta_msg").show();
         return;
     }
 
-    if(foundDuplicatedName(t, extra_meta_name, edit_extra_meta_row_id)) {
-        $("#extra_meta_msg").html("<font color='red'>The name already exists. Please input a different name.</font>");
+    if (extra_meta_name.length == 0 || extra_meta_value.length == 0) {
+        $("#extra_meta_msg").html("<div class='alert alert-danger'>" +
+            "Both name and value are required fields that cannot be left blank.</div>");
```

Comment:
@Maurier Is there also server-side checking for this as well?

Reply:
@aphelionz I disabled front end validations to test for this. I found that validation for empty required fields is not working properly. The ajax call returns an error, but the submission actually goes through.

Validations for unique constraints are working properly.

Reply:
@aphelionz and @Maurier I see this PR is targeted for current RC. Wondering why this feature needs to be part of the current RC? No new features should be added to any RC.

Reply:
Discussed this on the call, please leave in the RC